File: Python/WeightedWordMapping.py
class Solution(object):
    def mapWordWeights(self, words, weights):   
        result = []

        for word in words:
            total = 0

            for ch in word:
                total += weights[ord(ch) - ord('a')]

            total %= 26
            result.append(chr(ord('z') - total))

        return ''.join(result)

File: JS/WeightedWordMapping.js
var mapWordWeights = function(words, weights) {
    let result = "";

    for (const word of words) {
        let sum = 0;

        for (const ch of word) {
            sum += weights[ch.charCodeAt(0) - 'a'.charCodeAt(0)];
        }

        sum %= 26;
        result += String.fromCharCode('z'.charCodeAt(0) - sum);
    }

    return result;
};